libcxx/test/support/locale_helpers.h

```diff
@@ -80,6 +49,12 @@ std::wstring negate_en_US(std::wstring s) {
 #endif
 }
 
+wchar_t thousands_sep_or_default(std::wstring s) { return !s.empty() ? s[0] : L','; }
```

Comment:
You mean, that rather than the individual tests doing `const wchar_t fr_sep = LocaleHelpers::mon_thousands_sep_or_default(FR_MON_THOU_SEP);`, we'd just go `const wchar_t fr_sep = FR_MON_THOU_SEP;`?

I'm not entirely sure of the original author's intentions here, but I would guess that this felt like a smaller step - in case we didn't manage to dig up the right separators in the test framework. But as long as that setup does work (and afaik it does), I guess we could get rid of the defaults in `locale_helpers.h` entirely.

Reply:
> You mean, that rather than the individual tests doing `const wchar_t fr_sep = LocaleHelpers::mon_thousands_sep_or_default(FR_MON_THOU_SEP);`, we'd just go `const wchar_t fr_sep = FR_MON_THOU_SEP;`?

Yes, exactly!

Reply:
That doesn't work quite off the bat, at least.

The problem is that `FR_MON_THOU_SEP` is a wchar_t string like `L"\u00a0"`, while we want a single `wchar_t`. The `mon_thousands_sep_or_default` helper took a `std::wstring` and returns `s[0]` if `s` is nonempty in the current version of the patch.

This is because what `localeconv()` returns is a struct with strings, where each separator string usually is one single char, but they could in theory be multiple. In the current patch we treat this as strings all the way up to the `foo_sep_or_default()` helpers which convert from a string to a single wchar.

If we wanted to go that way, I guess we could make the `localeconv()` helper executable, that is called in python, only return the first char, and treat it as a numeric single wchar throughout instead. I guess that'd work too. It would be a bit tricky for one of the tests, though, where we currently try to do this:
```
    assert(stream_fr_FR_locale<CharT>(-1'000'000s) == L"-1" FR_THOU_SEP "000" FR_THOU_SEP "000s");
```

I'm not sure how we'd easily synthesize a wchar string literal out of that, when we'd have e.g. `-DFR_THOU_SEP=0x00a0`.

Reply:
Thanks @mstorsjo for taking over the PR and making progress on it!

Right, localeconv's struct has `currency_symbol` which could in practice be more than 1 char, so to allow `provide_locale_conversions` substitution code in features.py to be flexible enough it just sets the substitutions as strings.

The defaults ("," and ".") in the `thousands_sep_or_default` helper function won't actually be used as we are only calling these from tests where the locale is set to fr_FR or ru_RU and we've set the defines.

I think it's reasonable for these helpers to assert that the input is not empty and always take the first char. The functions should probably be renamed to something more appropriate though if they're essentially just converting to a wchar_t.